src/servers/http_server_v2.cc

```diff
@@ -501,54 +501,88 @@ HTTPAPIServerV2::ResponseRelease(
 template <typename T>
 void
 ReadDataFromJsonHelper(
-    T* data_vec, const DataType dtype, const rapidjson::Value& payload_data,
-    int* counter)
+    std::vector<T>* base, const DataType dtype,
+    const rapidjson::Value& payload_data, int* counter)
 {
   for (size_t i = 0; i < payload_data.Size(); i++) {
     // If last dimension
     if (!payload_data[i].IsArray()) {
       switch (dtype) {
-        case TYPE_BOOL:
+        case TYPE_BOOL: {
+          uint8_t* data_vec = reinterpret_cast<uint8_t*>(base->data());
           data_vec[*counter] = (uint8_t)payload_data[i].GetBool();
           break;
-        case TYPE_UINT8:
+        }
+        case TYPE_UINT8: {
+          uint8_t* data_vec = reinterpret_cast<uint8_t*>(base->data());
           data_vec[*counter] = (uint8_t)payload_data[i].GetInt();
           break;
-        case TYPE_UINT16:
+        }
+        case TYPE_UINT16: {
+          uint16_t* data_vec = reinterpret_cast<uint16_t*>(base->data());
           data_vec[*counter] = (uint16_t)payload_data[i].GetInt();
           break;
-        case TYPE_UINT32:
+        }
+        case TYPE_UINT32: {
+          uint32_t* data_vec = reinterpret_cast<uint32_t*>(base->data());
           data_vec[*counter] = (uint32_t)payload_data[i].GetInt();
           break;
-        case TYPE_UINT64:
+        }
+        case TYPE_UINT64: {
+          uint64_t* data_vec = reinterpret_cast<uint64_t*>(base->data());
           data_vec[*counter] = (uint64_t)payload_data[i].GetInt();
           break;
-        case TYPE_INT8:
+        }
+        case TYPE_INT8: {
+          int8_t* data_vec = reinterpret_cast<int8_t*>(base->data());
           data_vec[*counter] = (int8_t)payload_data[i].GetInt();
           break;
-        case TYPE_INT16:
+        }
+        case TYPE_INT16: {
+          int16_t* data_vec = reinterpret_cast<int16_t*>(base->data());
           data_vec[*counter] = (int16_t)payload_data[i].GetInt();
           break;
-        case TYPE_INT32:
+        }
+        case TYPE_INT32: {
+          int32_t* data_vec = reinterpret_cast<int32_t*>(base->data());
           data_vec[*counter] = (int32_t)payload_data[i].GetInt();
           break;
-        case TYPE_INT64:
+        }
+        case TYPE_INT64: {
+          int64_t* data_vec = reinterpret_cast<int64_t*>(base->data());
           data_vec[*counter] = (int64_t)payload_data[i].GetInt();
           break;
-        case TYPE_FP32:
+        }
+        case TYPE_FP32: {
+          float* data_vec = reinterpret_cast<float*>(base->data());
           data_vec[*counter] = (float)payload_data[i].GetFloat();
           break;
-        case TYPE_FP64:
+        }
+        case TYPE_FP64: {
+          double* data_vec = reinterpret_cast<double*>(base->data());
           data_vec[*counter] = (double)payload_data[i].GetDouble();
           break;
+        }
+        case TYPE_STRING: {
+          const char* cstr = payload_data[i].GetString();
+          uint32_t len = strlen(cstr);
```

Comment:
Shouldn't we be aware of the case where the content is not a C string? Say the serialized protobuf scenario. And if the rapidjson library be able to handle such case properly?

Reply:
@deadeyegoodwin can confirm this but we do not intend on supporting this through JSON. The same shall only be supported via the binary format method.

Reply:
Looking at www.json.org it actually looks like any byte array can likely be encoded in a json string using appropriate escapes. I would think that the rapidjson encoder/decoder should handle it. We will need to test thoroughly

Reply:
In the case that JSON encodes the byte array as a json string (assuming this is base64 encoded) it will still be null terminated as well right? 
Additionally if it is base64 encoded, we should have a flag to denote the same correct? So that we can decode it before passing to the backend.

Reply:
Nothing to do with base64. From the spec that I pointed to it appears that you can escape characters in the json string, so you should be able to escape a '\0' as well. You can feed some arbitrary byte arrays to radpijson and see how it encodes them into json.